schema/mysql/schema.sql

```diff
@@ -909,7 +909,9 @@ CREATE TABLE user_notification_history (
   notification_history_id binary(16) NOT NULL COMMENT 'UUID notification_history.id',
   user_id binary(20) NOT NULL COMMENT 'user.id',
 
-  PRIMARY KEY (id)
+  PRIMARY KEY (id),
+
+  CONSTRAINT fk_user_notification_history_notification_history FOREIGN KEY (notification_history_id) REFERENCES notification_history (id) ON DELETE CASCADE
```

Comment:
This constraint actually adds a dependency between the `icinga:history:stream:notification` and `icinga:history:stream:usernotification` streams. Thus, what I said in https://github.com/Icinga/icingadb/pull/259#issuecomment-895957599 is wrong and to support this schema change, we'd indeed have to change our code to wait for entries from other streams.